components/form/rangeDatepicker/src/index.js

```diff
@@ -0,0 +1,128 @@
+    }
+
+    this._handleChangeStart = this._handleChangeStart.bind(this)
+    this._handleChangeEnd = this._handleChangeEnd.bind(this)
+    this._handleClickButton = this._handleClickButton.bind(this)
+  }
+
+  _handleChangeStart (date) {
+    this.setState({
+      startDate: date
+    })
+    if (this.props.handleChangeStart) {
+      this.props.handleChangeStart(date)
+    }
+  }
+
+  _handleChangeEnd (date) {
+    this.setState({
+      endDate: date
+    })
+    if (this.props.handleChangeEnd) {
+      this.props.handleChangeEnd(date)
+    }
+  }
+
+  _handleClickButton () {
+    this.props.handleClickButton([this.state.startDate, this.state.endDate])
+  }
+
+  render () {
+    return (
+      <div className='sui-FormRangeDatepicker'>
+        <div className='sui-FormRangeDatepicker-item'>
+          <DatePicker
+            selected={this.state.startDate}
+            selectsStart
+            startDate={this.state.startDate}
+            endDate={this.state.endDate}
+            maxDate={moment(this.props.maxDate)}
+            onChange={this._handleChangeStart}
+            className='sui-FormRangeDatepicker-input'
+            locale={locale}
+          />
+          <div className='sui-FormRangeDatepicker-box'>
+            <Chevronbottom svgClass='sui-FormRangeDatepicker-icon' />
+          </div>
+        </div>
+        <div className='sui-FormRangeDatepicker-item'>
+          <DatePicker
+            selected={this.state.endDate}
+            selectsEnd
+            startDate={this.state.startDate}
+            endDate={this.state.endDate}
+            maxDate={moment(this.props.maxDate)}
+            onChange={this._handleChangeEnd}
+            className='sui-FormRangeDatepicker-input'
+            locale={locale}
+          />
+          <div className='sui-FormRangeDatepicker-box'>
+            <Chevronbottom svgClass='sui-FormRangeDatepicker-icon' />
+          </div>
+        </div>
+        {!!this.props.buttonLabel &&
+          <div className='sui-FormRangeDatepicker-button'>
+            <ButtonBasic
+              text={this.props.buttonLabel}
+              onClick={this._handleClickButton}
+            />
+          </div>
+        }
+      </div>
+    )
+  }
+}
+
+FormRangeDatepicker.displayName = 'FormRangeDatepicker'
+
+FormRangeDatepicker.propTypes = {
+  /**
+   * The string content is the label of the button.
+   * When is setted show a button.
+   */
+  buttonLabel: PropTypes.string,
+  /**
+   * End date of the selected range
+   */
+  endDate: PropTypes.object,
+  /**
+   * Event that will send the init date
+   */
+  handleChangeStart: PropTypes.func,
+  /**
+   * Event that will send the end date
+   */
+  handleChangeEnd: PropTypes.func,
+  /**
+   * Click event that will send the date range
+   */
+  handleClickButton: PropTypes.func,
+  /**
+   * Maximum date that the calendar allows to select
+   */
+  maxDate: PropTypes.object,
+  /**
+   * Init date of the selected range
+   */
```

Comment:
IMO date should be a string to ease implementation of component. But as React is JS, maybe I'm wrong privileging raw props...
You should document the expected date format in the comments of props types.
If you wanna expect a Date object, at least use PropTypes.instanceOf(Date)

Reply:
good! i like it